driver-core/src/main/java/com/datastax/driver/core/exceptions/CDCWriteException.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.driver.core.exceptions;
+
+import com.datastax.driver.core.EndPoint;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+
+/** An attempt was made to write to a commitlog segment which doesn't support CDC mutations * */
```

Comment:
Judging from the verbiage chosen server-side, it seems it has more to do with the CDC log folder being full:

```
String logMsg = String.format("Rejecting mutation to keyspace %s. Free up space in %s by processing CDC logs.",
                mutation.getKeyspaceName(), DatabaseDescriptor.getCDCLogLocation());
```

Reply:
My original comment was based on https://github.com/apache/cassandra/blob/trunk/src/java/org/apache/cassandra/db/commitlog/CommitLogSegmentManagerCDC.java#L130-L138.  As I look at the entirety of the comments + source there it is... quite confusing.  It seems like the comment in question literally contradicts itself.

I might just change this to some more generic language.

Reply:
Fixed